src/main/java/mate/academy/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,22 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.dao.UserDao;
+import mate.academy.lib.Inject;
+import mate.academy.model.User;
+import mate.academy.service.UserService;
+
+public class UserServiceImpl implements UserService {
+    @Inject
+    private UserDao userDao;
+
+    @Override
+    public User add(User user) {
+        return userDao.add(user);
```

Comment:
Ensure that the user's password is hashed before calling `userDao.add(user)`. Use `HashUtil.hashPassword` to hash the password and set it in the `User` object.